File: backend/google_login/jwt/src/main/java/com/security/jwt/repository/UserRepository.java
package com.security.jwt.repository;

import com.security.jwt.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.time.LocalDate;
import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    User findByUsername(String username);
    @Query("SELECT t FROM User t WHERE t.refreshtoken=:refresh")
    User exist (@Param("refresh") String refresh);
}

File: backend/user/src/main/kotlin/com/ssafy/leavelive/business/user/repository/UserRepository.kt
package com.ssafy.leavelive.business.user.repository

import com.ssafy.leavelive.business.user.model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, String> {
    fun existsByToken(token: String): Boolean
}